Flexa-Agents/Fallback-State/src/agent.ts

```diff
@@ -0,0 +1,53 @@
+import { BlockEvent, Finding, HandleBlock, FindingSeverity, FindingType, getEthersProvider } from "forta-agent";
+
+import DataFetcher from "./data.fetcher";
+
+const FLEXA_CONTRACT_ADDRESS = "0x706d7f8b3445d8dfc790c524e3990ef014e7c578";
+const FETCHER: DataFetcher = new DataFetcher(FLEXA_CONTRACT_ADDRESS, getEthersProvider());
+
+export const provideHandleBlock = (fetcher: DataFetcher, flexaContractAddress: string): HandleBlock => {
+  return async (blockEvent: BlockEvent): Promise<Finding[]> => {
+    const findings: Finding[] = [];
+
+    const previousBlockTimestamp = await fetcher.getPreviousBlockTimestamp(blockEvent.blockNumber);
+
+    const fallbackSetDate = await fetcher.getFallbackSetDate(blockEvent.blockNumber, flexaContractAddress);
+    const fallbackWithdrawalDelaySeconds = await fetcher.getFallbackWithdrawalDelaySeconds(
+      blockEvent.blockNumber,
+      flexaContractAddress
+    );
+    const previousFallbackSetDate = await fetcher.getFallbackSetDate(blockEvent.blockNumber - 1, flexaContractAddress);
+    const previousFallbackWithdrawalDelaySeconds = await fetcher.getFallbackWithdrawalDelaySeconds(
+      blockEvent.blockNumber - 1,
+      flexaContractAddress
+    );
```

Comment:
move this calls to inside the `if` in order to only do this queries when it is really needed.